forum/src/main/java/forum/controllers/PostController.java

```diff
@@ -0,0 +1,123 @@
+package forum.controllers;
+
+import forum.entities.Post;
+import forum.entities.Topic;
+import forum.entities.User;
+import forum.entities.Vote;
+import forum.services.*;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/posts")
+public class PostController {
+
+    private final PostService postService;
+    private final VoteService voteService;
+    private final UserService userService;
+    private final TopicService topicService;
+
+    public PostController(PostService postService,
+                          VoteService voteService,
+                          UserService userService,
+                          TopicService topicService) {
+        this.voteService = voteService;
+        this.postService = postService;
+        this.userService = userService;
+        this.topicService = topicService;
+    }
+
+    @GetMapping("{postId}")
+    public @ResponseBody
+    Post getPostByID(@PathVariable("postId") Long postId) {
+        return postService.findPostById(postId);
+    }
+
+    @PostMapping
+    public @ResponseBody
+    Post createNewPost(@RequestParam Long topicId, @RequestParam Long userId, @RequestBody String newPostText) {
+        Post newPostCreator = null;
+        User user = userService.findUserById(userId);
+        Topic topic = topicService.findById(topicId);
+        if (user != null && topic != null) {
+            newPostCreator = new Post();
+            newPostCreator.setAuthor(user);
+            newPostCreator.setTopic(topic);
+            newPostCreator.setText(newPostText);
+            newPostCreator.setDateCreated(LocalDateTime.now());
+            postService.save(newPostCreator);
+        }
+        return newPostCreator;
+    }
+
+    @PutMapping("/{postId}")
+    public @ResponseBody
+    Post updatePostById(@PathVariable("postId") Long postId, @RequestBody String newText) {
+        postService.updatePostById(postId, newText);
```

Comment:
Spring Data allows you to receive updated object after query execution. So, you don't need to make several calls.